packages/bridge/contracts/l1/L1DaiGateway.sol

```diff
@@ -195,4 +180,12 @@ contract L1DaiGateway is L1CrossDomainEnabled {
     // user encoded
     (maxSubmissionCost, extraData) = abi.decode(extraData, (uint256, bytes));
   }
+
+  function calculateL2TokenAddress(address l1Token) external view override returns (address) {
```

Comment:
This function can be made public and used to validate internally which tokens are handled by the gateway.

By that I mean, instead of
```solidity
require(l1Token == l1Dai, "L1DaiGateway/token-not-dai");
```
you could do

```solidity
require(calculateL2TokenAddress(l1Token) != address(0), "L1DaiGateway/token-not-dai");
```

This way the input validation logic always lives in the same place